azurerm/resource_arm_resource_group_test.go

```diff
@@ -25,29 +25,29 @@ func testSweepResourceGroups(region string) error {
 		return err
 	}
 
-	client := (*armClient).resourceGroupClient
+	client := (*armClient).resourceGroupsClient
+	ctx := (*armClient).StopContext
 
 	log.Printf("Retrieving the Resource Groups..")
-	results, err := client.List("", utils.Int32(int32(1000)))
+	results, err := client.List(ctx, "", utils.Int32(int32(1000)))
 	if err != nil {
 		return fmt.Errorf("Error Listing on Resource Groups: %+v", err)
 	}
 
-	for _, resourceGroup := range *results.Value {
+	for _, resourceGroup := range results.Values() {
 		if !shouldSweepAcceptanceTestResource(*resourceGroup.Name, *resourceGroup.Location, region) {
 			continue
 		}
 
 		name := *resourceGroup.Name
 		log.Printf("Deleting Resource Group %q", name)
-		deleteResponse, deleteErr := client.Delete(name, make(chan struct{}))
-		resp := <-deleteResponse
-		err = <-deleteErr
+		deleteFuture, err := client.Delete(ctx, name)
 		if err != nil {
-			if utils.ResponseWasNotFound(resp) {
-				continue
-			}
+			return err
+		}
 
+		err = deleteFuture.WaitForCompletion(ctx, client.Client)
```

Comment:
if `ctx` is done, should you skip this? will this method handle it?

Reply:
In looking through it, seems like it does listen on `Done`, so not an issue I don't think.

Reply:
indeed it does - it handles that within this method as part of the SDK (technically Go-AutoRest)